Mathlib/Analysis/SpecialFunctions/Log/Poslog.lean

```diff
@@ -0,0 +1,157 @@
+/-
+Copyright (c) 2025 Stefan Kebekus. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Stefan Kebekus
+-/
+import Mathlib.Analysis.SpecialFunctions.Log.Basic
+
+/-!
+# The Positive Part of the Logarithm
+
+This file defines the function `log⁺ = r ↦ max 0 (log r)`, and establishes
+standard estimates.
+-/
+
+open Real
+
+/-!
+## Definition, Notation and Reformulations
+-/
+/-- Definition: the real part of the logarithm. -/
+noncomputable def Real.poslog : ℝ → ℝ := fun r ↦ max 0 (log r)
+
+/-- Notation `log⁺` for the real part of the logarithm. -/
+notation "log⁺" => poslog
+
+/-- Definition of the real part of the logarithm, formulated as a theorem. -/
+theorem Real.poslog_def {r : ℝ} : log⁺ r = max 0 (log r) := rfl
+
+/-!
+## Elementary Properties
+-/
+/-- Presentation of `log` in terms of its positive part. -/
+theorem Real.log_eq_poslog_sub_poslog_inv {r : ℝ} : log r = log⁺ r - log⁺ r⁻¹ := by
+  rw [poslog_def, poslog_def, log_inv]
+  by_cases h : 0 ≤ log r
+  · simp [h]
+  · rw [not_le] at h
+    simp [neg_nonneg.1 (Left.nonneg_neg_iff.2 h.le)]
+
+/-- Presentation of `log⁺` in terms of `log`. -/
+theorem Real.poslog_eq_half_mul_log_add_log_abs {r : ℝ} : log⁺ r = 2⁻¹ * (log r + |log r|) := by
+  by_cases hr : 0 ≤ log r
+  · simp [poslog, hr, abs_of_nonneg]
+    ring
+  · simp [poslog, le_of_not_ge hr, abs_of_nonpos]
+
+/-- The positive part of `log` is never negative. -/
+theorem Real.poslog_nonneg {x : ℝ} : 0 ≤ log⁺ x := by simp [poslog]
+
+/-- The function `log⁺` is even. -/
+theorem Real.poslog_neg (x : ℝ) : log⁺ x = log⁺ (-x) := by simp [poslog]
```

Comment:
ditto. And you could make it a simp lemma.

Reply:
Implemented.